src/main/java/learnositysdk/request/Telemetry.java

```diff
@@ -0,0 +1,160 @@
+	private final static String OS_LINUX = "linux";
+	private final static String OS_WIN = "win";
+
+	static boolean isEnabled()
+	{
+		return telemetryEnabled;
+	}
+
+	static void disableTelemetry()
+	{
+		Telemetry.telemetryEnabled = false;
+	}
+
+	static void enableTelemetry()
+	{
+		Telemetry.telemetryEnabled = true;
+	}
+
+	/**
+	 * Add the telemetry data to the request packet, under "meta.sdk".
+	 *
+	 * @param requestPacket The request to add the telemetry data to
+	 */
+	static void addToRequest(JSONObject requestPacket)
+	{
+		JSONObject meta;
+
+		if (!Telemetry.telemetryEnabled) {
+			return;
+		}
+
+		try {
+			meta = requestPacket.getJSONObject("meta");
+			meta.put("sdk", Telemetry.getTelemetryData());
+		} catch (JSONException e) {
+			meta = new JSONObject();
+			meta.put("sdk", Telemetry.getTelemetryData());
+			requestPacket.put("meta", meta);
+		}
+	}
+
+	private static JSONObject getTelemetryData()
+	{
+		if (Telemetry.telemetryData != null) {
+			return Telemetry.telemetryData;
+		}
+
+		JSONObject telemetryData = Telemetry.telemetryData = new JSONObject();
+		telemetryData.put("version", Telemetry.getSdkVersion());
+		telemetryData.put("lang", Telemetry.getLanguage());
+		telemetryData.put("lang_version", System.getProperty("java.version"));
+		telemetryData.put("platform", Telemetry.getPlatform());
+		telemetryData.put("platform_version", Telemetry.getPlatformVersion());
+
+		return telemetryData;
+	}
+
+	private static String getSdkVersion()
+	{
+		String version = "unknown";
+		Properties properties = new Properties();
+		InputStream propertiesStream = Telemetry.class.getClassLoader().getResourceAsStream("project.properties");
+
+		if (propertiesStream != null) {
+			try {
+				properties.load(propertiesStream);
+				version = properties.getProperty("version");
+			} catch (Exception e) {
+				// Fail silently, returning default
+			}
+		}
+
+		return version;
+	}
+
+	private static String getLanguage()
+	{
+		String runtime = System.getProperty("java.runtime.name").toLowerCase(Locale.ENGLISH);
+
+		if (runtime.contains(Telemetry.RT_OPEN_JDK)) {
+			return Telemetry.RT_OPEN_JDK;
+		}
+
+		return Telemetry.RT_JAVA;
+	}
+
+	private static String getPlatform()
+	{
+		String os = System.getProperty("os.name").toLowerCase(Locale.ENGLISH);
+
+		if (os.contains("mac")) {
+			return Telemetry.OS_DARWIN;
+		}
+
+		if (os.contains("linux")) {
+			return Telemetry.OS_LINUX;
+		}
+
+		if (os.contains("windows")) {
+			return Telemetry.OS_WIN;
+		}
+
+		return os;
+	}
+
+	private static String getPlatformVersion()
+	{
+		// Attempt to manually fetch the kernel version for Darwin - java's OS version returns macOS version instead
+		if (Telemetry.getPlatform().equals(Telemetry.OS_DARWIN)) {
+			String version = Telemetry.unameVersion();
+
+			if (version != null) return version;
+			// Drop through if there was an issue getting the version from uname
+		}
+
+		return System.getProperty("os.version");
```

Comment:
What's the default for this?

Reply:
On linux and windows it returns the same as `php_uname('r')` as we want. The override for Mac is so we return the darwin kernel version instead of the OS X version.